articles/connections/social/oauth2.md

```diff
@@ -16,121 +16,106 @@ useCase:
-
-### Fetch user profile for OIDC-conformant OAuth2 providers
+The `user_id` property in the returned profile is required and the `email` property is optional but highly recommended. See the [User Profile Root Attributes reference](/users/updating-user-profile-root-attributes) for more information on  what attributes can be returned.
 
-If the OAuth2 provider is OIDC-conformant, such as another Auth0 tenant, you can extract the user profile from the ID Token. The `ctx` object in the fetch user profile script will have the ID Token. You can then decode the token and map its fields to the user object. For example:
-
-```js
-function(access_token, ctx, callback) {
-  var jwt = require("jsonwebtoken");
-  const idToken = jwt.decode(ctx.id_token);
-  const profile = {
-    "user_id": idToken.sub,
-    "name": idToken.name,
-    "given_name": idToken.given_name,
-    "family_name": idToken.family_name,
-    "middle_name": idToken.middle_name,
-    "nickname": idToken.nickname,
-    "preferred_username": idToken.preferred_username,
-    "profile": idToken.profile,
-    "picture": idToken.picture,
-    "website": idToken.website,
-    "email": idToken.email,
-    "email_verified": idToken.email_verified,
-    "gender": idToken.gender,
-    "birthdate": idToken.birthdate,
-    "zoneinfo": idToken.zoneinfo,
-    "locale": idToken.locale,
-    "phone_number": idToken.phone_number,
-    "phone_number_verified": idToken.phone_number_verified,
-    "address": idToken.address,
-    "updated_at": idToken.updated_at
-  };
-  console.log(JSON.stringify(profile, null, 2));
-  callback(null, profile);
-}
-```
+You can filter, add or remove anything from the profile returned from the provider. However, it is recommended that you keep this script as simple as possible. More sophisticated manipulation of user information can be achieved through the use of [Rules](/rules). One advantage of using Rules is that they apply to *any* connection.
 
 ## Log in using the custom connection
 
-You can use any of the Auth0 standard mechanisms (such as direct links, <dfn data-key="lock">[Auth0 Lock](/libraries/lock)</dfn>, [auth0.js](/libraries/auth0js), and so on.) to login a user with the your custom connection.
-
-A direct link would look like:
+You can use any of the Auth0 standard mechanisms to login a user with the your custom connection. A direct link would look like:
 
 ```text
 https://${account.namespace}/authorize
   ?response_type=code
   &client_id=${account.clientId}
   &redirect_uri=${account.callback}
-  &scope=openid%20profile
+  &scope=openid%20profile%20email
   &connection=THE_NAME_OF_THE_CONNECTION
-  &state=OPAQUE_VALUE
 ```
 
-To add a custom connection in Lock, you can add a custom button by following the instructions at [Adding custom connections to lock](/libraries/lock/v9/ui-customization#adding-a-new-ui-element-using-javascript) and using this link as the button `href`.
+## Modify the icon and display name
```

Comment:
These fields will be added to the custom form soon:

https://auth0team.atlassian.net/browse/DXEC-147